core/controllers/image/controller.php

```diff
@@ -3,12 +3,64 @@
 
 // api style controller - end output
 ob_end_clean();
-
+ob_end_clean(); // not a mistake, a safety net
 // router
 
 $segments = CMS::Instance()->uri_segments;
 $segsize = sizeof($segments);
 
+// handle list api request
+if ($segments[1]=='list_images') {
+	$mimetypes = array_filter(explode(',',Input::getvar('mimetypes','STRING'))) ?? null;
+	$searchtext = Input::getvar('searchtext','STRING');
+	$images_per_page = Input::getvar('images_per_page','INT') ?? 50;
+	$page = Input::getvar('page','INT') ?? 1;
+	if ($searchtext=='null') {
+		$searchtext=null;
+	}
+	if ($searchtext) {
+		$query = "SELECT * FROM `media` WHERE `title` LIKE ? OR alt LIKE ?";
+		if ($mimetypes) {
+			$query.=" AND mimetype IN (";
+			for ($n=0; $n<sizeof($mimetypes); $n++) {
+				if ($n>0) {
+					$query .= ",";
+				}
+				$query .= CMS::Instance()->pdo->quote($mimetypes[$n]);
+			}
+			$query.=")";
+		}
+		$query.=" LIMIT " . $images_per_page . " OFFSET " . ($page-1)*$images_per_page;
+		$list = DB::fetchAll($query, ["%$searchtext%","%$searchtext%"]);
+	}
+	else {
+		$query = "SELECT * FROM `media`";
+		if ($mimetypes) {
+			$query.=" WHERE id>0 ";
+		}
+		if ($mimetypes) {
+			// TODO: ensure valid mimetypes from JSON?
+			$query.=" AND mimetype in (";
+			for ($n=0; $n < sizeof($mimetypes); $n++) {
+				if ($n>0) {
+					$query .= ",";
+				}
+				$query .= CMS::Instance()->pdo->quote($mimetypes[$n]);
+			}
```

Comment:
foreach says hello from php 4. arguably code be converted down to an array function + implode